### ISSUE TITLE
Miscellaneous improvements

### ISSUE THREAD

**Issue by oscbyspro** [oscbyspro/DiffableTextViews#151]
I’m rather content with the state of the project, but there are a couple of things I’d like to improve, but that are not significant enough to warrant separate issues. So, instead, I’ll collect them here.

**Comment by oscbyspro** [comment]
### Pattern where capacity == 0 (aka the useless pattern)

The `hidden()` method should hide all characters, rather than hiding none. It behaves like this because the reduction function looks for the first placeholder characters, does not find one, then stops at the end index (meaning that the effect is the same as the first placeholder being the end index, which is false). Solving this should also remove the need for special case selection on 100% formatting.

**Comment by oscbyspro** [comment]
In Xcode 14.0 beta 5, updating the value while the view is active (with a button, etc) now yields:

```
⚠︎ Publishing changes from within view updates is not allowed, this will cause undefined behavior.
```

I don’t know if the inner workings of SwiftUI has changed of if the warning was previously suppressed (as I haven’t seen it in a while). It’s trivial to fix however, I’ll just change change `lock.perform` to `lock.task` in `push(_:)` to do the update async (after some testing to see that it’s OK).

**Comment by oscbyspro** [comment]
Apparently, I forgot to override `Snapshot/distance(from:to:)` so it’s still O(n).

**Comment by oscbyspro** [comment]
### Lazy views improve readability

It’s easier to reason about the complexity of code when laziness is explicit. So I think it would be better if `snapshot.nonvirtuals` where instead `snapshot.lazy.nonvirtuals()`. With two extensions, you could write:

```
snapshot.nonvirtuals()
snapshot.lazy.nonvirtuals()
```

A follow up to this (and an actual improvement) would be adding a `Proposal.Lazy` with a `merged()` method:

```
proposal.merged().nonvirtuals()
proposal.lazy.merged().nonvirtuals() // this option does not yet exist
```

**Comment by oscbyspro** [comment]
I think that `PatternTextStyle/resolve` can use the same reduce function as `format` and `interpret`. At least it looked like it before the Xcode 14.0 beta 5 decided it was time to become a pyromaniac. Anyways, it would be more performant and less code to maintain.

**Comment by oscbyspro** [comment]
Fixed a bug in 5940f6bd16709b2bb999fa24f9d65a8dc4996bd2 where localizable styles marked as `equals(())` would not be localized on setup. It’s not that note worthy, however, because the bug had not been introduced in outside of beta.

**Comment by oscbyspro** [comment]
Removed `Lock` in cfa97c681793bf7f9a268539da6d997b0f3b8a1c because it was made obsolete #161

**Comment by oscbyspro** [comment]
Cleaned up debug messages in ac76804ad8d98cd573adb115d04a84419d38ad28 as:

```
 [DiffableTextViews] [context]: [message]
```

**Comment by oscbyspro** [comment]
Fixed #161 `suggestion` + `autocorrection` + `reentrance` selection flash in cb8eb6529abdbadaa2262acc044de3de14dcf2cf.

**Comment by oscbyspro** [comment]
`PatternTextStyle`: made state public.
`PatternTextStyle`: added initializer parameters. 
`PatternTextPlaceholders`: made `ExpressibleByDictionaryLiteral`.
